utils/ips/ip_port.go

```diff
@@ -12,7 +12,44 @@ import (
 	"github.com/ava-labs/avalanchego/utils/wrappers"
 )
 
-var errBadIP = errors.New("bad ip format")
+const nullStr = "null"
+
+var (
+	errMissingQuotes = errors.New("first and last characters should be quotes")
+	errBadIP         = errors.New("bad ip format")
+)
+
+type IPDesc (IPPort)
+
+func (ipDesc IPDesc) String() string {
+	return IPPort(ipDesc).String()
+}
+
+func (ipDesc IPDesc) MarshalJSON() ([]byte, error) {
```

Comment:
Inconsistent pointer/value receivers w/ `UnmarshalJson`, can we make all of them the same type

Reply:
Added a unit test to show why it is needed.